recount2/1-get_all_recount_dataset.R

```diff
@@ -50,19 +52,26 @@ included.sample.list <- unique(metadata.nonempty$project)
 
 # Download all recount2 samples in included.sample.list
 lapply(included.sample.list, 
-       function(x) download_study(x, type = "rse-gene", outdir = file.path(data.dir, x)))
+       function(x) download_study(x, type = "rse-gene", 
+                                  outdir = file.path(data.dir, x)))
 
-rpkm <- data.frame()
-
-for(i in 1:length(included.sample.list)) {
-  rse.gene <- get(load(file.path(data.dir, included.sample.list[i], 'rse_gene.Rdata')))
-  if(i == 1) {
-    rpkm <- as.data.frame(getRPKM(rse.gene))
-    rpkm$id <- rownames(rpkm)
-  } else {
-    rpkm.tmp <- as.data.frame(getRPKM(rse.gene))
-    rpkm.tmp$id <- rownames(rpkm.tmp)
-    rpkm <- merge(rpkm, rpkm.tmp, by=c("id"))
-  }
+# get RPKM for each experiment and add to list
+rpkm.list <- list()
+for(experiment in included.sample.list) {
+  load(file.path(data.dir, experiment, 'rse_gene.Rdata'))
+  rpkm <- as.data.frame(getRPKM(rse_gene))
+  rpkm$id <- rownames(rpkm)
+  rpkm.list[[experiment]] <- rpkm
 }
-save(rpkm, file = "recount.rpkm.RData")
+
+# combine experiments -- this is the most memory efficient way to go about this
+# that I've found -- will need to drop extraneous gene id columns
+rpkm.df <- do.call(base::cbind, c(rpkm.list, by = "id"))
```

Comment:
does [`dplyr::bind_cols()`](http://dplyr.tidyverse.org/reference/bind.html) work?

Reply:
Did a bit of exploration of this topic here: https://github.com/greenelab/rheum-plier-data/pull/10/files#diff-336c115ba63149bca524a54de200d2b2R214

I believe this will be more efficient